src/app/helpers/filters/audioRecordingFilters.ts

```diff
@@ -0,0 +1,78 @@
+import { DateExpressions, InnerFilter } from "@baw-api/baw-api.service";
+import { DateTime } from "luxon";
+import { AudioRecording } from "@models/AudioRecording";
+import { addFilter } from "./filters";
+
+/**
+ * Adds date range conditions to an existing filter
+ *
+ * @param startDate (optional) The minimum `recordedDate` of the date range
+ * @param endDate (optional) The maximum `recordedEndDate` allowed
+ * @param filters (optional) An existing filter to add the date range conditions to. If no filter is provided, a new filter will be created
+ * @returns A new filter with all the same conditions as the initial filter, with the date range conditions added
+ */
+export function addDateFilters(
+  startDate?: DateTime,
+  endDate?: DateTime,
+  filters?: InnerFilter
+): InnerFilter {
+  if (startDate) {
+    // the baw-api is expecting the date in the format (YYYY-MM-DD)
+    // ISO date format is used as it emits the date in the correct format and explicitly emits the time as 00:00:00
+    const formattedStartDate = startDate.toISO();
+    const startDateFilter: InnerFilter<AudioRecording> = {
+      recordedDate: { greaterThan: formattedStartDate },
+    };
+
+    filters = addFilter(filters, startDateFilter);
+  }
+
+  if (endDate) {
+    const formattedEndDate = endDate.toISO();
+    const endDateFilters = {
+      recordedEndDate: { lessThanOrEqual: formattedEndDate },
+    };
+
+    filters = addFilter(filters, endDateFilters);
+  }
+
+  return filters;
+}
+
+/**
+ * Adds time range conditions to an existing filter
+ *
+ * @param expressions Describes if the time filters use local (daylight savings) time
+ * @param startTime (optional) The minimum `recordedDate` allowed
+ * @param endTime (optional) The maximum `recordedEndDate` allowed
+ * @param filters (optional) An existing filter to add the time range conditions to. If no filter is provided, a new filter will be created
+ * @returns
+ */
+export function addTimeFilters(
+  expressions: DateExpressions[],
+  startTime?: string,
+  endTime?: string,
```

Comment:
weird that these are formatted as string when the other method takes a well-typed construct (`DateTime`) though I see luxon has no time only type... soo I guess there's no action to take here


maybe validate input value is time by regex??

Reply:
Time input is already validated through angular form validation on the `time.component.ts`, and is completely detached from the filter component

Reply:
You miss the point: many well-typed structures serialize well by default. For example the Luxon DateTime object implements toJSON and emits an ISO8601 value automatically and also normalizes the input date.

This has nothing to do with validation.

Reply:
so start time and and time should be luxon durations i think